src/services/services.ts

```diff
@@ -1306,39 +1376,8 @@ namespace ts {
             if (host.useSourceOfProjectReferenceRedirect) {
                 compilerHost.useSourceOfProjectReferenceRedirect = () => host.useSourceOfProjectReferenceRedirect!();
             }
-            host.setCompilerHost?.(compilerHost);

```

Comment:
A lot of fairly uninteresting refactors through this section that creates a big diff. The compiler host creation was moved into a function so it can be used both for the main program and for the auto import provider. Small tweaks within some of the host functions were necessary to ensure, e.g., that we don’t release source files from one program that are still being used by the other, and that ScriptInfo creation knows whether it’s being created as a real part of the project or just as an auto-importable file.

Reply:
I think instead of just moving compilerHost into the new function you should move program creation and  setting hostCache to undefined to avoid using it later on and getting typechecker to ensure that files are bound and have correct parent pointers..  here as well.. So that its one unit that does everything right for getting you program